testutil/environment/environment.go

```diff
@@ -167,6 +167,27 @@ func (e *Execution) IsRootless() bool {
 	return os.Getenv("DOCKER_ROOTLESS") != ""
 }
 
+// IsUserNamespaceInKernel returns whether the kernel supports user namespaces
+func (e *Execution) IsUserNamespaceInKernel() bool {
```

Comment:
Wondering; was `IsUserNamespace()` not working for this case?

Reply:
IIUC `IsUserNamespace()` checks if we are running with a user-namespaced daemon right now. Whereas for the test I've written I'm interested in knowing if it's possible to run a user-namespaced daemon in the current test environment.

Reply:
Ah, right.

Was also wondering if there wasn't an existing utility for this, but looks indeed that the existing ones check if it's _enabled_ (not if it's _supported_); looks like the daemon does not actually check for that; https://github.com/moby/moby/blob/cf31b9622ae9de4b5fb7a1c9456460980ab80281/daemon/daemon_unix.go#L1167-L1170 (perhaps something we should add)

For reference; this was copied from the equivalent utility in `integration-cli`; https://github.com/moby/moby/blob/46cdcd206c56172b95ba5c77b827a722dab426c5/integration-cli/requirements_test.go#L136-L139

( we should look at moving those remaining tests to `integration` as well)